src/core/ext/filters/client_channel/subchannel.cc

```diff
@@ -919,9 +918,9 @@ void Subchannel::OnRetryAlarm(void* arg, grpc_error* error) {
   if (error == GRPC_ERROR_NONE) {
     gpr_log(GPR_INFO, "Failed to connect to channel, retrying");
     c->ContinueConnectingLocked();
-    gpr_mu_unlock(&c->mu_);
+    lock.Unlock();
   } else {
-    gpr_mu_unlock(&c->mu_);
+    lock.Unlock();
```

Comment:
The need to use `ReleasableMutexLock` instead of just `MutexLock` here will probably go away with the work that @AspirinSJL is doing to simplify subchannel refcounting.  Once that's done, we can use `RefCountedPtr<>` here, call its `release()` method in the case where we call `ContinueConnectingLocked()`, and otherwise let it unref at destruction after the lock has been released.

Can you please add a TODO about this?

Reply:
That'd be really great. Sure, added a TODO here.